src/server/routes/signIn.ts

```diff
@@ -325,24 +356,31 @@ router.post(
   }),
 );
 
-const handleAppRedirect = (
-  res: ResponseWithRequestState,
-  returnUrl?: string,
-): void => {
-  if (returnUrl) {
-    const url = new URL(decodeURIComponent(returnUrl));
-    // we check to see if the return url has a custom scheme
-    // which lets us know the request has come from the app
-    // so we can redirect back to the app
-    if (
-      validAppProtocols.includes(url.protocol) &&
-      validateReturnUrl(returnUrl) !== defaultReturnUri
-    ) {
-      return res.redirect(returnUrl);
-    }
+type SocialProvider = 'google' | 'facebook' | 'apple';
+
+router.get('/signin/:social', (req: Request, res: ResponseWithRequestState) => {
+  // todo can we use the login middleware here to stopped logged in users from accessing this endpoint?
+  const { useOkta, returnUrl } = res.locals.queryParams;
+  const socialIdp = req.params.social as SocialProvider;
+
+  // validate social is one of facebook, google, or apple
+  // if not, redirect to signin
+  if (!['facebook', 'google', 'apple'].includes(socialIdp)) {
```

Comment:
what d'you think about having a var called `isValidSocialNetwork` or similar? I think we could avoid the small two line comment above with a name like that

Reply:
created a `isValidSocialProvider` helper method